pkg/oc/generate/app/cmd/resolve.go

```diff
@@ -254,7 +268,7 @@ func DetectSource(repositories []*app.SourceRepository, d app.Detector, g *Gener
 				errs = append(errs, errors.New("No Jenkinsfile was found in the repository and the requested build strategy is 'pipeline'"))
 			}
 		default:
-			if repo.Info().Dockerfile == nil && !repo.Info().Jenkinsfile && len(repo.Info().Types) == 0 {
+			if repo.Info().Dockerfile == nil && !repo.Info().Jenkinsfile && len(repo.Info().Types) == 0 && git.IsGitInstalled() {
```

Comment:
not clear why you're calling isgitinstalled here.   DetectSource should not be called at all if we couldn't get the source code.  If we can get the source code (either because git is installed, or the source is in a local dir) then detection should proceed as normal.